src/dataframe/dataframe.jl

```diff
@@ -646,7 +639,7 @@ function insertcols!(df::DataFrame, col_ind::Int, name_col::Pair{Symbol, <:Abstr
                      makeunique::Bool=false)
     name, item = name_col
     if !(0 < col_ind <= ncol(df) + 1)
-        throw(BoundsError("attempt to insert a column to a data frame with $(ncol(df)) columns at index $col_ind"))
+        throw(DimensionMismatch("attempt to insert a column to a data frame with $(ncol(df)) columns at index $col_ind"))
```

Comment:
I'd just use an `ArgumentError`: the docstring for `DimensionMismatch` says "The objects called do not have matching dimensionality", which isn't the case here.
```suggestion
        throw(ArgumentError("attempt to insert a column to a data frame with $(ncol(df)) columns at index $col_ind"))
```